geode-core/src/distributedTest/java/org/apache/geode/internal/cache/ClearPRDuringGIIRequesterDUnitTest.java

```diff
@@ -0,0 +1,339 @@
+
+    waitForGIITeskHookStarted(memberVMS[2], giiTestHookType);
+
+    if (clearBeforeGII) {
+      for (int i = 1; i < memberVMS.length; i++) {
+        memberVMS[i].invoke(() -> {
+          PauseDuringClearDistributionMessageObserver observer =
+              (PauseDuringClearDistributionMessageObserver) DistributionMessageObserver
+                  .getInstance();
+          DistributionMessageObserver.setInstance(null);
+          observer.latch.countDown();
+        });
+      }
+
+      memberVMS[2].invoke(() -> {
+        InitialImageOperation.resetGIITestHook(giiTestHookType, true);
+      });
+    } else {
+      memberVMS[2].invoke(() -> {
+        InitialImageOperation.resetGIITestHook(giiTestHookType, true);
+      });
+
+      for (int i = 1; i < memberVMS.length; i++) {
+        memberVMS[i].invoke(() -> {
+          PauseDuringClearDistributionMessageObserver observer =
+              (PauseDuringClearDistributionMessageObserver) DistributionMessageObserver
+                  .getInstance();
+          DistributionMessageObserver.setInstance(null);
+          observer.latch.countDown();
+        });
+      }
+    }
+
+    try {
+      asyncGII.join(10000);
+    } catch (InterruptedException ex) {
+      Assert.fail("Async recreate region interupted" + ex.getMessage());
+    }
+    try {
+      asyncClear.join(10000);
+    } catch (InterruptedException ex) {
+      Assert.fail("Async clear interupted" + ex.getMessage());
+    }
+
+    if (!clearBeforeGII && asyncClear.exceptionOccurred()) {
+      assertThat(asyncClear.getException() instanceof PartitionedRegionPartialClearException);
+    } else {
+      verifyRegionSizes(0);
+    }
+  }
+
+  @Test
+  public void clearBeforeGIIShouldClearTheRegion() {
+    giiTestHookSyncWithClear(true);
+  }
+
+  @Test
+  public void clearAfterGIIShouldClearTheRegion() {
+    giiTestHookSyncWithClear(false);
+  }
+
+  private void verifyRegionSizes(int expectedSize) {
+    for (int i = 2; i < memberVMS.length; i++) {
+      memberVMS[i].invoke(() -> {
+        PartitionedRegion pr = (PartitionedRegion) getCache().getRegion(REGION_NAME);
+        for (BucketRegion br : pr.getDataStore().getAllLocalBucketRegions()) {
+          logger.info("verifyRegionSizes:" + br.getFullPath() + ":"
+              + br.getBucketAdvisor().isPrimary() + ":" + br.size());
+        }
+      });
+    }
+    for (int i = 1; i < memberVMS.length; i++) {
+      memberVMS[i].invoke(() -> verifyRegionSize(expectedSize));
+    }
+  }
+
+  public void waitForGIITeskHookStarted(final MemberVM vm,
+      final InitialImageOperation.GIITestHookType callbacktype) {
+    SerializableRunnable waitForCallbackStarted = new SerializableRunnable() {
+      @Override
+      public void run() {
+
+        final InitialImageOperation.GIITestHook callback =
+            getGIITestHookForCheckingPurpose(callbacktype);
+        WaitCriterion ev = new WaitCriterion() {
```

Comment:
`WaitCriterion` is deprecated and should not be used. Instead, `GeodeAwaitility.await()` should be used to assert the conditions, e.g. 
```
GeodeAwaitility.await().untilAsserted(() -> assertThat(getGIITestHookForCheckingPurpose(callbacktype)).satisfies(callback -> {
          assertThat(callback).isNotNull();
          assertThat(callback.isRunning).isTrue();
        }));
```